open-hackathon-server/src/hackathon/storage/cryptor.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+"""
+Copyright (c) Microsoft Open Technologies (Shanghai) Co. Ltd.  All rights reserved.
+
+The MIT License (MIT)
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+from hashlib import md5
+from Crypto.Cipher import AES
+from Crypto import Random
+from urllib2 import urlopen
+import os
+
+from hackathon import Component
+
+__all__ = ["Cryptor"]
+
+
+class Cryptor(Component):
+
+    def __init__(self):
+        self.password = self.util.safe_get_config("storage.azure.account_key", "123456")
+
+    def derive_key_and_iv(self, salt, key_length, iv_length):
+        d = d_i = ''
+        while len(d) < key_length + iv_length:
+            d_i = md5(d_i + self.password + salt).digest()
+            d += d_i
+        return d[:key_length], d[key_length:key_length + iv_length]
+
+
+    def encrypt(self, in_file_name, out_file_name, key_length=32):
+        in_file = open(in_file_name, 'rb')
+        out_file = open(out_file_name, 'wb')
+        bs = AES.block_size
+        salt = Random.new().read(bs - len('Salted__'))
+        key, iv = self.derive_key_and_iv(salt, key_length, bs)
+        cipher = AES.new(key, AES.MODE_CBC, iv)
+        out_file.write('Salted__' + salt)
+        finished = False
+        while not finished:
+            chunk = in_file.read(1024 * bs)
+            if len(chunk) == 0 or len(chunk) % bs != 0:
+                padding_length = (bs - len(chunk) % bs) or bs
+                chunk += padding_length * chr(padding_length)
+                finished = True
+            out_file.write(cipher.encrypt(chunk))
```

Comment:
Please handle Exceptions and logging as well. Usually byte and file operations easily fail. And also the caller may want to know the result of the encrypt/decrypt methods. Is encryption success or not?